src/main/views/MattermostBrowserView.ts

```diff
@@ -368,8 +368,8 @@ export class MattermostBrowserView extends EventEmitter {
     }
 
     private handleTitleUpdate = (e: Event, title: string) => {
-        this.log.debug('handleTitleUpdate', title);
-
+        this.log.info('handleTitleUpdate', title);
+        MainWindow.get()?.setTitle('Mattermost Desktop - ' + title);
```

Comment:
This should reflect the app's name in case the app is whitelabelled. Unfortunately right now most people will just override the title in the `webpack.config.renderer.js` file, so we need to think about how to best capture that.

We could make a call to `app.getName()` which currently returns `Mattermost` for most builds, but that would be a change from what it currently is.

@asaadmahmood Any thoughts about what this could look like?

Reply:
@devinbinnie I think I'm lacking context here. I'm not exactly sure what you're asking us. Can you elaborate a bit more about what aspect of the Window title you want UX input on?

Reply:
@matthewbirtch Sorry about that. What we're looking to refine is the window title of the application. So for example, the prompt you see here on Window, when you hover the taskbar icon:
![image](https://github.com/mattermost/desktop/assets/52460000/aacde787-6d20-4553-bce7-1f140625e334)

You can also see it on Linux as the native title bar:
![image](https://github.com/mattermost/desktop/assets/52460000/7743f65a-0cf4-4fdb-940a-aa54cf34cae0)

I don't think you can see it on macOS, but I could be wrong.

Reply:
Not sure it needs to include 'Desktop App' - I would just say 'Mattermost'.

But from the PR description, it looks like you're hoping to incorporate the selected server, team and channel as well? 

I would propose `Mattermost - {Channel Name} - {Team Name} - {Server Name}`

That feels like a lot to include in a window title though. Maybe we can omit the Team?

Reply:
Please note that the pull request, in its current form, just forwards the window title from the server application, whatever that may set, as the app title prefixed by the application's own application title. Trying to slim it down would either require a deeper interfacing between the electron wrapper and the server webapp or relying on specific title formats, both of which would introduce a dependency of which I'm not sure whether would be desirable.

Also, the application title is derived from the configured application title by capturing the initial title setting in order to support whitelabeling as per @devinbinnie 's objection here: https://github.com/mattermost/desktop/pull/2719#discussion_r1193807677. So the change from "Mattermost Desktop App" to just "Mattermost" would require changing the title in `webpack.config.renderer.js`, however, I'm unsure about the ramifications this would have.

Reply:
@matthewbirtch I'll let you have the final say here on what we want the title to look like. I'm definitely more on board for just having it say `Mattermost` instead of `Mattermost Desktop App`.

@hennejg I'm happy to take a stab at helping finish up this PR if you need assistance getting the title right. Again, I think we can avoid changing things in the Webpack file if we just use the call I mentioned above.

Reply:
> @hennejg I'm happy to take a stab at helping finish up this PR if you need assistance getting the title right. Again, I think we can avoid changing things in the Webpack file if we just use the call I mentioned above.

I am sorry, but I still confused about what we are trying to achieve here. I was under the impression that deriving the window title from what is configured in `webpack.config.renderer.js` (via the `HtmlWebpackPlugin`'s `title`) is the desired behaviour, e.g. to allow while-labeling. Over there, the title is configured to be `Mattermost Desktop App`, though. If it should still be just `Mattermost` and a change to `webpack.config.renderer.js` isn't the right way to achieve that, it would certainly be trivial to replace `Mattermost Desktop App` with `Mattermost` programmatically.

Reply:
@hennejg Whitelabelling usually involving changing the applications name in `package.json`. Currently under `productName` ours is `Mattermost`, but if you change that then `app.name` changes in Electron as well. This is what we should be using for whitelabelling, not the Webpack file. The webpack file only provides a default, like how you would hardcode the `<title>` tag.

We want to be able to programmatically override this, and add additional information as well.

Reply:
Ok, I think now we're on the same page. I updated the PR to derive the title prefix from `app.name`. However, unfortunately I am unable to test this right now, because `app.name` doesn't work in dev mode but only in packaged mode. And I can't easily package due to my very weird VS setup (I am mostly doing embedded development).